src/ILLink.RoslynAnalyzer/ReflectionMethodBodyScanner.cs

```diff
@@ -91,6 +108,37 @@ public static bool HandleCall (OperationAnalysisContext context, IInvocationOper
 			var calledMethod = invocationOperation.TargetMethod;
 			switch (GetIntrinsicIdForMethod (calledMethod)) {
 
+				//
+				// GetMethod (string)
+				// GetMethod (string, BindingFlags)
+				// GetMethod (string, Type[])
+				// GetMethod (string, Type[], ParameterModifier[])
+				// GetMethod (string, BindingFlags, Type[])
+				// GetMethod (string, BindingFlags, Binder, Type[], ParameterModifier[])
+				// GetMethod (string, BindingFlags, Binder, CallingConventions, Type[], ParameterModifier[])
+				// GetMethod (string, int, Type[])
+				// GetMethod (string, int, Type[], ParameterModifier[]?)
+				// GetMethod (string, int, BindingFlags, Binder?, Type[], ParameterModifier[]?)
+				// GetMethod (string, int, BindingFlags, Binder?, CallingConventions, Type[], ParameterModifier[]?)
+				//
+				case IntrinsicId.Type_GetMethod: {
+						BindingFlags? bindingFlags;
+						// @TODO - only handling the first 2 cases here
+						if (calledMethod.Parameters.Length > 1 && calledMethod.Parameters[1].Type.GetDisplayName () == "System.Reflection.BindingFlags")
+							bindingFlags = GetBindingFlagsFromValue (invocationOperation.Arguments[1]);
+						else
+							// Assume a default value for BindingFlags for methods that don't use BindingFlags as a parameter
+							bindingFlags = BindingFlags.Public;
+
+						DynamicallyAccessedMemberTypes memberTypes = GetDynamicallyAccessedMemberTypesFromBindingFlagsForMethods (bindingFlags);
+
+						var source = DynamicallyAccessedMembersAnalyzer.TryGetSymbolFromOperation (invocationOperation.Instance, context);
+						if (source != null)
+							DynamicallyAccessedMembersAnalyzer.VerifyAnnotations (source.Value, memberTypes, context, invocationOperation.Syntax.GetLocation ());
```

Comment:
My BindingLogic in VerifyAnnotations is limited, for intrinsics it shouldn't be operating with the DynamicallyAccessedMembersTypes that we got from BindingFlags, it should operate directly with the binding flags since that's what the binder in Linker does. But either way, I don't think the linker will print nor keep anything with BindingFlags.CreateInstance, since is an ignored flag.
Also, I talked with Sven about this branch he suggested two things that I think are valuable, one is to divide the tasks into something more like the trimmer. Instead of filtering members and getting diagnostics in a single method, have two methods one that returns an enumeration of members (it will behave like the marking in trimmer) and another that only generates diagnostics.
The second recommendation was to divide this PR, Binding Logic + Intrinsics is an extensive change especially considering the new dataflow changes which make both tasks more complex.

Reply:
I'm a little behind our goal :-( Pushing a fix to handle intrinsic method separately. Still partial - BindingFlags are not fully handled and need to take care of RUC, DynamicDependency, delegates and type extension annotations (need to investigate if this can be handled in the mainline).

Will have a plan to discuss at our sync up today.